### ISSUE TITLE
Cannot find Adobe Key

### ISSUE THREAD

**Issue by l-l-l-l-l-l** [apprenticeharper/DeDRM_tools#1251]
I can't manage to get the Adobe Key neither directy with the plugin nor by executing adobekey.py by itself.

What are the steps I have to do to show you a log of the issue?

**Comment by ElleKayEm** [comment]
You have Adobe Digital Editions authorized with an Adobe ID and installed on the same computer and user as calibre with DeDRM?  Which version of ADE are you using?

Here's how to get a log:  https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md#i-cannot-solve-my-problem-with-the-dedrm-plugin-and-now-i-need-to-post-a-log-how-do-i-do-that

**Comment by l-l-l-l-l-l** [comment]
Yes, yes and both 2.0.1.78765 and 4.5 (removed authorization from 4.5 and reauthorized with 2.0)

log from a PDF:
`Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 703, in initialize_plugins
  File "site-packages\calibre\customize\ui.py", line 61, in load_plugin
  File "site-packages\calibre\customize\zipplugin.py", line 202, in load
  File "importlib\__init__.py", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 618, in _load_backward_compatible
  File "site-packages\calibre\customize\zipplugin.py", line 184, in load_module
  File "calibre_plugins.dummy1.__init__", line 9, in <module>
ModuleNotFoundError: No module named 'urlparse'
Traceback (most recent call last):
  File "site-packages\calibre\gui2\ui.py", line 157, in __init__
  File "site-packages\calibre\gui2\ui.py", line 171, in init_iaction
  File "site-packages\calibre\customize\__init__.py", line 614, in load_actual_plugin
  File "importlib\__init__.py", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 618, in _load_backward_compatible
  File "site-packages\calibre\customize\zipplugin.py", line 184, in load_module
  File "calibre_plugins.obok_dedrm.action", line 24, in <module>
  File "site-packages\calibre\customize\zipplugin.py", line 184, in load_module
  File "calibre_plugins.obok_dedrm.dialogs", line 30, in <module>
  File "site-packages\calibre\customize\zipplugin.py", line 184, in load_module
  File "calibre_plugins.obok_dedrm.utilities", line 10, in <module>
ModuleNotFoundError: No module named 'StringIO'
EpubMerge: DEBUG: 2020-09-27 20:34:34,482: calibre_plugins.epubmerge.epubmerge_plugin(158): macmenuhack file_path:C:\Users\Jorge\AppData\Roaming\calibre\plugins\fanficfare_macmenuhack.txt
calibre Registro de depuración
calibre 5.0.1 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.18362-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.18362')
Python 3.8.5
Windows: ('10', '10.0.18362', 'SP0', 'Multiprocessor Free')
Interface language: es
Failed to initialize plugin: 'C:\\Users\\Jorge\\AppData\\Roaming\\calibre\\plugins\\Kindle hi-res covers.zip'
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 34, 0) && DeDRM (7, 0, 0) && Package KFX (from KFX Input) (1, 34, 0) && DOC Input (1, 0, 1) && Embed Comic Metadata (1, 5, 0) && EpubMerge (2, 11, 0) && KFX metadata reader (from KFX Input) (1, 34, 0) && KFX Input (1, 34, 0) && KePub Input (3, 2, 10) && KePub Metadata Reader (3, 2, 9) && Obok DeDRM (6, 7, 0) && ZIP-RAR to Comic Book (1, 0, 0)
calibre 5.0.1 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.18362-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.18362')
Python 3.8.5
Windows: ('10', '10.0.18362', 'SP0', 'Multiprocessor Free')
Interface language: es
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 34, 0) && DeDRM (7, 0, 0) && Package KFX (from KFX Input) (1, 34, 0) && DOC Input (1, 0, 1) && Embed Comic Metadata (1, 5, 0) && EpubMerge (2, 11, 0) && KFX metadata reader (from KFX Input) (1, 34, 0) && KFX Input (1, 34, 0) && KePub Input (3, 2, 10) && KePub Metadata Reader (3, 2, 9) && Obok DeDRM (6, 7, 0) && ZIP-RAR to Comic Book (1, 0, 0)
Not controlling automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 102.23899371069182 x 102.35820895522387
Using calibre Qt style: True
[0.00] Starting up...
[0.00] Showing splash screen...
[0.69] splash screen shown
[0.69] Initializing db...
[0.73] db initialized
[0.73] Constructing main UI...
Failed to check for calibre update: [WinError 10013] Intento de acceso a un socket no permitido por sus permisos de acceso
Failed to check for plugin update: [WinError 10013] Intento de acceso a un socket no permitido por sus permisos de acceso
[1.59] main UI initialized...
[1.59] Hiding splash screen
Starting QuickView
[8.09] splash screen hidden
[8.09] Started up in 8.09 seconds with 863 books

Debugging DND event
application/x-qt-windows-mime;value="Shell IDList Array" 314 b'\x01\x00\x00\x00\x0c\x00\x00\x00\xd6\x00\x00\x00\x14\x00\x1fP\xe0O\xd0 \xea:i\x10\xa2\xd8\x08\x00+00\x9d:\x00.\x80\x92+\x16\xd3e\x93zF\x95k\x92p:\xca\x08\xaf&\x00\x01\x00&\x00\xef\xbe\x11\x00\x00\x00\x9dh\xcc\x13{f\xd6\x01\xbd?\x05\x1b\x98\x84\xd6\x01V[:g\xfcp\xd6\x01\x14\x00z\x001\x00\x00\x00\x00\x00;Q\x89p\x10\x00My Digital Editions\x00X\x00\t\x00\x04\x00\xef\xbe;Qak;Q\x89p.\x00\x00\x00\xa1\xb1\x00\x00\x00\x00\x07\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x17\xd8\xbf\x00M\x00y\x00 \x00D\x00i\x00g\x00i\x00t\x00a\x00l\x00 \x00E\x00d\x00i\x00t\x00i\x00o\x00n\x00s\x00\x00\x00"\x00\x00\x00b\x002\x006\x88\x10\x00;Q\x84p \x00Dracula.pdf\x00H\x00\t\x00\x04\x00\xef\xbe;Q\x84p;Q\x84p.\x00\x00\x00\xb9\xb1\x00\x00\x00\x00\x06\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00r\xad#\x00D\x00r\x00a\x00c\x00u\x00l\x00a\x00' 

application/x-qt-windows-mime;value="UsingDefaultDragImage" 4 b'\x01\x00\x00\x00' 

application/x-qt-windows-mime;value="DragImageBits" 36888 b'`\x00\x00\x00`\x00\x00\x000\x00\x00\x00Y\x00\x00\x00-*\x05\xd2\xff\xff\xff\xff\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' 

application/x-qt-windows-mime;value="DragContext" 16 b'\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' 

application/x-qt-windows-mime;value="DragSourceHelperFlags" 4 b'\x01\x00\x00\x00' 

application/x-qt-windows-mime;value="InShellDragLoop" 4 b'\x01\x00\x00\x00' 

text/uri-list 58 b'file:///D:/Documents/My%20Digital%20Editions/Dracula.pdf\r\n' 

application/x-qt-windows-mime;value="FileName" 45 b'D:\\Documents\\My Digital Editions\\Dracula.pdf\x00' 

application/x-qt-windows-mime;value="FileContents" 0 b'' 

application/x-qt-windows-mime;value="FileNameW" 90 b'D\x00:\x00\\\x00D\x00o\x00c\x00u\x00m\x00e\x00n\x00t\x00s\x00\\\x00M\x00y\x00 \x00D\x00i\x00g\x00i\x00t\x00a\x00l\x00 \x00E\x00d\x00i\x00t\x00i\x00o\x00n\x00s\x00\\\x00D\x00r\x00a\x00c\x00u\x00l\x00a\x00.\x00p\x00d\x00f\x00\x00\x00' 

application/x-qt-windows-mime;value="FileGroupDescriptorW" 596 b'\x01\x00\x00\x00l\x00\x00\x80\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00 \x00\x00\x00\x00\xfc\x1c\x11\xd7\x94\xd6\x01\x00\x00\x00\x00\x00\x00\x00\x00\x8eN\xf9\x10\xd7\x94\xd6\x01\x00\x00\x00\x006\x88\x10\x00D\x00r\x00a\x00c\x00u\x00l\x00a\x00.\x00p\x00d\x00f\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' 

application/x-qt-windows-mime;value="IsShowingLayered" 4 b'\x01\x00\x00\x00' 

application/x-qt-windows-mime;value="DragWindow" 4 b'\x1e\x0eE\x00' 

application/x-qt-windows-mime;value="IsComputingImage" 4 b'\x00\x00\x00\x00' 

application/x-qt-windows-mime;value="DropDescription" 1044 b'\xff\xff\xff\xff\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' 

application/x-qt-windows-mime;value="DisableDragText" 4 b'\x01\x00\x00\x00' 

application/x-qt-windows-mime;value="IsShowingText" 4 b'\x00\x00\x00\x00' 

application/x-qt-windows-mime;value="ComputedDragImage" 4 b'-*\x05\xd2' 


Firefox rurl: None None
URLS: ['file:///D:/Documents/My%20Digital%20Editions/Dracula.pdf']
Paths: ['D:/Documents/My Digital Editions/Dracula.pdf']
Extensions: frozenset({'pdf'})
DeDRM v7.0.0: Trying to decrypt Dracula.pdf
DeDRM v7.0.0: Dracula.pdf is a PDF ebook
DeDRM v7.0.0: Looking for new default Adobe Digital Editions Keys after 0.4 seconds
DeDRM v7.0.0: Exception when getting default Adobe Key after 0.4 seconds
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 454, in PDFDecrypt
  File "calibre_plugins.dedrm.adobekey", line 360, in adeptkeys
  File "calibre_plugins.dedrm.adobekey", line 340, in CryptUnprotectData
calibre_plugins.dedrm.adobekey.ADEPTError: Failed to decrypt user key key (sic)
DeDRM v7.0.0: Ultimately failed to decrypt after 0.4 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 633, in run
  File "calibre_plugins.dedrm.__init__", line 508, in PDFDecrypt
calibre_plugins.dedrm.DeDRMError: DeDRM v7.0.0: Ultimately failed to decrypt after 0.4 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 428, in get_file_type_metadata
  File "site-packages\calibre\customize\builtins.py", line 343, in get_metadata
  File "site-packages\calibre\ebooks\metadata\pdf.py", line 127, in get_metadata
ValueError: Could not read info dict from PDF
Failed to initialize plugin: 'C:\\Users\\Jorge\\AppData\\Roaming\\calibre\\plugins\\Kindle hi-res covers.zip'
Syntax Error: Couldn't find the 'EBX_HANDLER' security handler

pdfinfo errored out with return code: 1
Exception ignored in: <function NativeFunction.__del__ at 0x000001DF9BE49DC0>
Traceback (most recent call last):
  File "calibre_plugins.dedrm.adobekey", line 266, in __del__
TypeError: 'NoneType' object is not callable
Exception ignored in: <function NativeFunction.__del__ at 0x000001DF9BE49DC0>
Traceback (most recent call last):
  File "calibre_plugins.dedrm.adobekey", line 266, in __del__
TypeError: 'NoneType' object is not Saume:  book 1542 added with 1 formats.
Saume:  checking format pdf for book 1542.
Saume:  book 1542 added with 1 formats.
Saume:  checking format pdf for book 1542.
Added Dracula to db in: 1.5
Added 1 books in 3.4 seconds
`

log from epub:
`Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 703, in initialize_plugins
  File "site-packages\calibre\customize\ui.py", line 61, in load_plugin
  File "site-packages\calibre\customize\zipplugin.py", line 202, in load
  File "importlib\__init__.py", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 618, in _load_backward_compatible
  File "site-packages\calibre\customize\zipplugin.py", line 184, in load_module
  File "calibre_plugins.dummy1.__init__", line 9, in <module>
ModuleNotFoundError: No module named 'urlparse'
Traceback (most recent call last):
  File "site-packages\calibre\gui2\ui.py", line 157, in __init__
  File "site-packages\calibre\gui2\ui.py", line 171, in init_iaction
  File "site-packages\calibre\customize\__init__.py", line 614, in load_actual_plugin
  File "importlib\__init__.py", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 618, in _load_backward_compatible
  File "site-packages\calibre\customize\zipplugin.py", line 184, in load_module
  File "calibre_plugins.obok_dedrm.action", line 24, in <module>
  File "site-packages\calibre\customize\zipplugin.py", line 184, in load_module
  File "calibre_plugins.obok_dedrm.dialogs", line 30, in <module>
  File "site-packages\calibre\customize\zipplugin.py", line 184, in load_module
  File "calibre_plugins.obok_dedrm.utilities", line 10, in <module>
ModuleNotFoundError: No module named 'StringIO'
EpubMerge: DEBUG: 2020-09-27 20:37:43,560: calibre_plugins.epubmerge.epubmerge_plugin(158): macmenuhack file_path:C:\Users\Jorge\AppData\Roaming\calibre\plugins\fanficfare_macmenuhack.txt
calibre Registro de depuración
calibre 5.0.1 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.18362-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.18362')
Python 3.8.5
Windows: ('10', '10.0.18362', 'SP0', 'Multiprocessor Free')
Interface language: es
Failed to initialize plugin: 'C:\\Users\\Jorge\\AppData\\Roaming\\calibre\\plugins\\Kindle hi-res covers.zip'
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 34, 0) && DeDRM (7, 0, 0) && Package KFX (from KFX Input) (1, 34, 0) && DOC Input (1, 0, 1) && Embed Comic Metadata (1, 5, 0) && EpubMerge (2, 11, 0) && KFX metadata reader (from KFX Input) (1, 34, 0) && KFX Input (1, 34, 0) && KePub Input (3, 2, 10) && KePub Metadata Reader (3, 2, 9) && Obok DeDRM (6, 7, 0) && ZIP-RAR to Comic Book (1, 0, 0)
calibre 5.0.1 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.18362-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.18362')
Python 3.8.5
Windows: ('10', '10.0.18362', 'SP0', 'Multiprocessor Free')
Interface language: es
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 34, 0) && DeDRM (7, 0, 0) && Package KFX (from KFX Input) (1, 34, 0) && DOC Input (1, 0, 1) && Embed Comic Metadata (1, 5, 0) && EpubMerge (2, 11, 0) && KFX metadata reader (from KFX Input) (1, 34, 0) && KFX Input (1, 34, 0) && KePub Input (3, 2, 10) && KePub Metadata Reader (3, 2, 9) && Obok DeDRM (6, 7, 0) && ZIP-RAR to Comic Book (1, 0, 0)
Not controlling automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 102.23899371069182 x 102.35820895522387
Using calibre Qt style: True
[0.00] Starting up...
[0.00] Showing splash screen...
[0.72] splash screen shown
[0.72] Initializing db...
[0.75] db initialized
[0.75] Constructing main UI...
Failed to check for calibre update: [WinError 10013] Intento de acceso a un socket no permitido por sus permisos de acceso
Failed to check for plugin update: [WinError 10013] Intento de acceso a un socket no permitido por sus permisos de acceso
[1.55] main UI initialized...
[1.55] Hiding splash screen
Starting QuickView
[4.44] splash screen hidden
[4.44] Started up in 4.44 seconds with 863 books

Debugging DND event
application/x-qt-windows-mime;value="Shell IDList Array" 396 b'\x01\x00\x00\x00\x0c\x00\x00\x00\xd6\x00\x00\x00\x14\x00\x1fP\xe0O\xd0 \xea:i\x10\xa2\xd8\x08\x00+00\x9d:\x00.\x80\x92+\x16\xd3e\x93zF\x95k\x92p:\xca\x08\xaf&\x00\x01\x00&\x00\xef\xbe\x11\x00\x00\x00\x9dh\xcc\x13{f\xd6\x01\xbd?\x05\x1b\x98\x84\xd6\x01V[:g\xfcp\xd6\x01\x14\x00z\x001\x00\x00\x00\x00\x00;Q\x89p\x10\x00My Digital Editions\x00X\x00\t\x00\x04\x00\xef\xbe;Qak;Q\x89p.\x00\x00\x00\xa1\xb1\x00\x00\x00\x00\x07\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x17\xd8\xbf\x00M\x00y\x00 \x00D\x00i\x00g\x00i\x00t\x00a\x00l\x00 \x00E\x00d\x00i\x00t\x00i\x00o\x00n\x00s\x00\x00\x00"\x00\x00\x00\xb4\x002\x00+\x01\x04\x00;Q\x82p \x00The Adventures of Sherlock Holmes.epub\x00\x00~\x00\t\x00\x04\x00\xef\xbe;Q\x82p;Q\x82p.\x00\x00\x00\xb5\xb1\x00\x00\x00\x00\x04\x00\x00\x00\x00\x00' 

application/x-qt-windows-mime;value="UsingDefaultDragImage" 4 b'\x01\x00\x00\x00' 

application/x-qt-windows-mime;value="DragImageBits" 36888 b'`\x00\x00\x00`\x00\x00\x000\x00\x00\x00Y\x00\x00\x00\\&\x05\x1b\xff\xff\xff\xff\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' 

application/x-qt-windows-mime;value="DragContext" 16 b'\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' 

application/x-qt-windows-mime;value="DragSourceHelperFlags" 4 b'\x01\x00\x00\x00' 

application/x-qt-windows-mime;value="InShellDragLoop" 4 b'\x01\x00\x00\x00' 

text/uri-list 93 b'file:///D:/Documents/My%20Digital%20Editions/The%20Adventures%20of%20Sherlock%20Holmes.epub\r\n' 

application/x-qt-windows-mime;value="FileName" 72 b'D:\\Documents\\My Digital Editions\\The Adventures of Sherlock Holmes.epub\x00' 

application/x-qt-windows-mime;value="FileContents" 0 b'' 

application/x-qt-windows-mime;value="FileNameW" 144 b'D\x00:\x00\\\x00D\x00o\x00c\x00u\x00m\x00e\x00n\x00t\x00s\x00\\\x00M\x00y\x00 \x00D\x00i\x00g\x00i\x00t\x00a\x00l\x00 \x00E\x00d\x00i\x00t\x00i\x00o\x00n\x00s\x00\\\x00T\x00h\x00e\x00 \x00A\x00d\x00v\x00e\x00n\x00t\x00u\x00r\x00e\x00s\x00 \x00o\x00f\x00 \x00S\x00h\x00e\x00r\x00l\x00o\x00c\x00k\x00 \x00H\x00o\x00l\x00m\x00e\x00s\x00.\x00e\x00p\x00u\x00b\x00\x00\x00' 

application/x-qt-windows-mime;value="FileGroupDescriptorW" 596 b'\x01\x00\x00\x00l\x00\x00\x80\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00 \x00\x00\x00\x00\xa2\xba\x0e\xd7\x94\xd6\x01\x00\x00\x00\x00\x00\x00\x00\x00\xb5\x0fr\x0e\xd7\x94\xd6\x01\x00\x00\x00\x00+\x01\x04\x00T\x00h\x00e\x00 \x00A\x00d\x00v\x00e\x00n\x00t\x00u\x00r\x00e\x00s\x00 \x00o\x00f\x00 \x00S\x00h\x00e\x00r\x00l\x00o\x00c\x00k\x00 \x00H\x00o\x00l\x00m\x00e\x00s\x00.\x00e\x00p\x00u\x00b\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' 


Firefox rurl: None None
URLS: ['file:///D:/Documents/My%20Digital%20Editions/The%20Adventures%20of%20Sherlock%20Holmes.epub']
Paths: ['D:/Documents/My Digital Editions/The Adventures of Sherlock Holmes.epub']
Extensions: frozenset({'epub'})
DeDRM v7.0.0: Trying to decrypt The Adventures of Sherlock Holmes.epub
DeDRM v7.0.0: Verifying zip archive integrity
DeDRM v7.0.0: Error 'a bytes-like object is required, not 'str'' when checking zip archive
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 203, in ePubDecrypt
  File "calibre_plugins.dedrm.zipfix", line 50, in __init__
  File "calibre_plugins.dedrm.zipfilerugged", line 683, in __init__
  File "calibre_plugins.dedrm.zipfilerugged", line 709, in _GetContents
  File "calibre_plugins.dedrm.zipfilerugged", line 752, in _RealGetContents
  File "calibre_plugins.dedrm.zipfilerugged", line 289, in __init__
TypeError: a bytes-like object is required, not 'str'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 637, in run
  File "calibre_plugins.dedrm.__init__", line 207, in ePubDecrypt
Exception: a bytes-like object is required, not 'str'
Failed to initialize plugin: 'C:\\Users\\Jorge\\AppData\\Roaming\\calibre\\plugins\\Kindle hi-res covers.zipSaume:  book 1543 added with 1 formats.
Saume:  checking format epub for book 1543.
Saume:  book 1543 added with 1 formats.
Saume:  checking format epub for book 1543.
Added The Adventures of Sherlock Holmes to db in: 1.4
Added 1 books in 2.9 seconds
`

**Comment by ElleKayEm** [comment]
Well, assuming this isn't related to using calibre 5 and the pre-release DeDRM 7...

You can read your books in ADE 2.0.1?  There may have been some kind of glitch with de-authorizing ADE 4.5 and authorizing ADE 2.  See TROUBLESHOOTING REAUTHORIZATION AND DEAUTHORIZATION ISSUES ON WINDOWS on this page:

https://helpx.adobe.com/digital-editions/kb/activation-errors-adobe-digital-editions.html

These are instructions for deleting a registry key.  As it says there, always be extra cautious when editing Windows registry.

**Comment by l-l-l-l-l-l** [comment]
Yes, I can read them (after deleting the original authorization and reathorizing it on ADE 2.0, which as I said, I did at the start).

Most likely due to the prerelease, then. I hope those logs help in some way to polish the files for release.

PS. I wonder if there's some way to gete the key from the xml files ADE leaves on authorized devices.

**Comment by ElleKayEm** [comment]
Okay, sometimes taking the extra step of deleting the registry key, then re-authorizing clears things up.

**Comment by l-l-l-l-l-l** [comment]
Okay, I did the registry reset.
Now when I try to get the adobe key, it runs until it tells me to name it, then fails due to an error on line 300 of config.py

Ran adobekey.py by itself and got a key file. I imported it, but still didn't decrypt.

Log for the new attempt (only relevant parts):

> Firefox rurl: None None
URLS: ['file:///D:/Documents/My%20Digital%20Editions/Dracula.pdf', 'file:///D:/Documents/My%20Digital%20Editions/El%20ingenioso%20hidalgo%20Don%20Quijote%20de%20la%20Mancha.pdf', 'file:///D:/Documents/My%20Digital%20Editions/Historia%20de%20la%20vida%20del%20Busc%C3%B3n.epub', 'file:///D:/Documents/My%20Digital%20Editions/The%20Adventures%20of%20Sherlock%20Holmes.epub']
Paths: ['D:/Documents/My Digital Editions/Dracula.pdf', 'D:/Documents/My Digital Editions/El ingenioso hidalgo Don Quijote de la Mancha.pdf', 'D:/Documents/My Digital Editions/Historia de la vida del Buscón.epub', 'D:/Documents/My Digital Editions/The Adventures of Sherlock Holmes.epub']
Extensions: frozenset({'pdf', 'epub'})
DeDRM v7.0.0: Trying to decrypt Dracula.pdf
DeDRM v7.0.0: Dracula.pdf is a PDF ebook
DeDRM v7.0.0: Trying to decrypt El ingenioso hidalgo Don Quijote de la Mancha.pdf
DeDRM v7.0.0: El ingenioso hidalgo Don Quijote de la Mancha.pdf is a PDF ebook
DeDRM v7.0.0: Trying to decrypt Historia de la vida del Buscón.epub
DeDRM v7.0.0: Verifying zip archive integrity
DeDRM v7.0.0: Error 'a bytes-like object is required, not 'str'' when checking zip archive
DeDRM v7.0.0: Trying to decrypt The Adventures of Sherlock Holmes.epub
DeDRM v7.0.0: Verifying zip archive integrity
DeDRM v7.0.0: Error 'a bytes-like object is required, not 'str'' when checking zip archive
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 633, in run
  File "calibre_plugins.dedrm.__init__", line 423, in PDFDecrypt
AttributeError: 'str' object has no attribute 'decode'
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 428, in get_file_type_metadata
  File "site-packages\calibre\customize\builtins.py", line 343, in get_metadata
  File "site-packages\calibre\ebooks\metadata\pdf.py", line 127, in get_metadata
ValueError: Could not read info dict from PDF
Failed to initialize plugin: 'C:\\Users\\Jorge\\AppData\\Roaming\\calibre\\plugins\\Kindle hi-res covers.zip'
Syntax Error: Couldn't find the 'EBX_HANDLER' security handler
> 
> pdfinfo errored out with return code: 1
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 633, in run
  File "calibre_plugins.dedrm.__init__", line 423, in PDFDecrypt
AttributeError: 'str' object has no attribute 'decode'
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 428, in get_file_type_metadata
  File "site-packages\calibre\customize\builtins.py", line 343, in get_metadata
  File "site-packages\calibre\ebooks\metadata\pdf.py", line 127, in get_metadata
ValueError: Could not read info dict from PDF
Failed to initialize plugin: 'C:\\Users\\Jorge\\AppData\\Roaming\\calibre\\plugins\\Kindle hi-res covers.zip'
Syntax Error: Couldn't find the 'EBX_HANDLER' security handler
> 
> pdfinfo errored out with return code: 1
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 203, in ePubDecrypt
  File "calibre_plugins.dedrm.zipfix", line 50, in __init__
  File "calibre_plugins.dedrm.zipfilerugged", line 683, in __init__
  File "calibre_plugins.dedrm.zipfilerugged", line 709, in _GetContents
  File "calibre_plugins.dedrm.zipfilerugged", line 752, in _RealGetContents
  File "calibre_plugins.dedrm.zipfilerugged", line 289, in __init__
TypeError: a bytes-like object is required, not 'str'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 637, in run
  File "calibre_plugins.dedrm.__init__", line 207, in ePubDecrypt
Exception: a bytes-like object is required, not 'str'
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 203, in ePubDecrypt
  File "calibre_plugins.dedrm.zipfix", line 50, in __init__
  File "calibre_plugins.dedrm.zipfilerugged", line 683, in __init__
  File "calibre_plugins.dedrm.zipfilerugged", line 709, in _GetContents
  File "calibre_plugins.dedrm.zipfilerugged", line 752, in _RealGetContents
  File "calibre_plugins.dedrm.zipfilerugged", line 289, in __init__
TypeError: a bytes-like object is required, not 'str'
> 
>During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 637, in run
  File "calibre_plugins.dedrm.__init__", line 207, in ePubDecrypt
Exception: a bytes-like object is required, not 'str'
Failed to initialize plugin: 'C:\\Users\\Jorge\\AppData\\Roaming\\calibre\\plugins\\Kindle hi-res covSaume:  book 1547 added with 1 formats.
Saume:  checking format epub for book 1547.
Saume:  book 1547 added with 1 formats.
Saume:  checking format epub for book 1547.
Added Historia de la vida del Buscón to db in: 1.5
Saume:  book 1548 added with 1 formats.
Saume:  checking format pdf for book 1548.
Saume:  book 1548 added with 1 formats.
Saume:  checking format pdf for book 1548.
Added Dracula to db in: 1.4
Saume:  book 1549 added with 1 formats.
Saume:  checking format pdf for book 1549.
Saume:  book 1549 added with 1 formats.
Saume:  checking format pdf for book 1549.
Added El ingenioso hidalgo Don Quijote de la Mancha to db in: 1.4
Saume:  book 1550 added with 1 formats.
Saume:  checking format epub for book 1550.
Saume:  book 1550 added with 1 formats.
Saume:  checking format epub for book 1550.
Added The Adventures of Sherlock Holmes to db in: 1.6
Added 4 books in 7.7 seconds

**Comment by ElleKayEm** [comment]
Alright, I think it's due to DeDRM 7 not being quite ready yet.  So you can either wait on DeDRM 7's release or roll back to calibre 4.23 and DeDRM 6.8.

**Comment by l-l-l-l-l-l** [comment]
I'll wait for it, since I don't really need it that urgently.

I hope those logs help with the proper upgrade to Python 3.

PS. I was also checking Obok, but when I click on the customization button it tells me that I need to restart calibre before that. Every time.

**Comment by ElleKayEm** [comment]
Yes, Obok also needs updating for Python 3, and I imagine it will take a backseat to DeDRM.

**Comment by l-l-l-l-l-l** [comment]
Speaking about Obok has reminded me of another ticket I have to make.
